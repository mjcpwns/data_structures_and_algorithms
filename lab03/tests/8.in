1 10 4 2 67 8 3 5 31 7 6 77 64 9 44
