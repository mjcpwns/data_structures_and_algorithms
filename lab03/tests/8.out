Rotating left
Rotating left
Rotating left
Rotating left
Rotating right
Rotating left
Rotating left
Rotating left
Rotating left
Rotating left
Rotating right
Tree:
          7
         / \
        /   \
       /     \
      /       \
     /         \
    4          31
   / \         / \
  2   5       /   \
 / \   \     /     \
1   3   6   9      67
           / \     / \
          8  10   /   \
                 64   77
                /
               44

#nodes = 15,  #leaves = 7
Infix  : 1 2 3 4 5 6 7 8 9 10 31 44 64 67 77 
Prefix : 7 4 2 1 3 5 6 31 9 8 10 67 64 44 77 
Postfix: 1 3 2 6 5 4 8 10 9 44 64 77 67 31 7 
ByLevel: 7 4 31 2 5 9 67 1 3 6 8 10 64 77 44 
